system_metrics_collector/src/system_metrics_collector/periodic_measurement_node.hpp

```diff
@@ -93,6 +93,20 @@ class PeriodicMeasurementNode : public system_metrics_collector::Collector,
   rclcpp_lifecycle::node_interfaces::LifecycleNodeInterface::CallbackReturn on_error(
     const rclcpp_lifecycle::State & previous_state);
 
+  /**
+   * Return the name of the metric collected.
+   *
+   * @return string reporesenting name of the collected metric
+   */
+  std::string GetMetricName() const override = 0;
```

Comment:
This is really confusing. PeriodicMeasurementNode inherits from both Collector and MetricsMessagePublisher.

Reply:
That is why [Virtual Inheritance](https://en.wikipedia.org/wiki/Virtual_inheritance) is used for disambiguation.